src/Bicep.LangServer/BicepCompilationManager.cs

```diff
@@ -269,7 +269,7 @@ private ImmutableArray<ISourceFile> CloseCompilationInternal(DocumentUri documen
                     diagnostics = diagnostics.Append(configurationDiagnostic);
                 }
 
-                if (version == 1)
+                if (triggeredByFileOpenEvent)
                 {
                     SendTelemetryOnBicepFileOpen(context.Compilation.GetEntrypointSemanticModel(), documentUri.ToUri(), configuration, sourceFiles, diagnostics);
```

Comment:
Instead of having triggeredByFileOpenEvent param, can the SendTelemetryOnBicepFileOpen code be called directly by the file open handler?

Reply:
???

Reply:
Thanks for the suggestion Stephen!

I don't recollect the reason why that code was initially put in BicepCompilationManager and not in BicepTextDocumentSyncHandler. We do post more than just file open information- https://github.com/Azure/bicep/blob/3a7eafd3da47e9f6ac2611333a91276df48b559a/src/Bicep.LangServer/BicepCompilationManager.cs#L309. BicepTextDocumentSyncHandler has access to ICompilationManager so we could retrieve all the information, I need to make sure there's no rework/perf impact.  I Created https://github.com/Azure/bicep/issues/7856 for further investigation.

Reply:
k sounds like it's not worth it then. thx